data-prepper-plugins/otel-trace-source/src/main/java/com/amazon/dataprepper/plugins/certificate/acm/ACMCertificateProvider.java

```diff
@@ -28,42 +25,41 @@
 import java.nio.ByteBuffer;
 import java.security.PrivateKey;
 import java.security.Security;
+import java.util.Optional;
+import java.util.UUID;
 
-public class ACMCertificateProvider {
+public class ACMCertificateProvider implements CertificateProvider {
     private static final Logger LOG = LoggerFactory.getLogger(ACMCertificateProvider.class);
     private static final long SLEEP_INTERVAL = 10000L;
     private static final String BOUNCY_CASTLE_PROVIDER = "BC";
     private final AWSCertificateManager awsCertificateManager;
+    private final String acmArn;
     private final long totalTimeout;
-
-    public ACMCertificateProvider(final String awsRegion, final long totalTimeout) {
-        final AWSCredentialsProvider credentialsProvider = new DefaultAWSCredentialsProviderChain();
-        final ClientConfiguration clientConfig = new ClientConfiguration()
-                .withThrottledRetries(true);
-        this.awsCertificateManager = AWSCertificateManagerClientBuilder.standard()
-            .withRegion(awsRegion)
-            .withCredentials(credentialsProvider)
-            .withClientConfiguration(clientConfig)
-            .build();
-        this.totalTimeout = totalTimeout;
-        Security.addProvider(new BouncyCastleProvider());
-    }
-
-    // accessible only in the same package for unit test
-    ACMCertificateProvider(final AWSCertificateManager awsCertificateManager, final long totalTimeout) {
+    private final String passphrase;
+    public ACMCertificateProvider(final AWSCertificateManager awsCertificateManager,
+                                  final String acmArn,
+                                  final long totalTimeout,
+                                  final String passphrase) {
         this.awsCertificateManager = awsCertificateManager;
+        this.acmArn = acmArn;
```

Comment:
Similar to other comment, constructors of new classes can make use of `Objects.requireNonNull()`

Reply:
Will add this.